app/controllers/twitter-searcher.controller.js

```diff
@@ -0,0 +1,130 @@
+(function() {
+  'use strict';
+
+  class TwitterSearcherCtrl {
+    constructor($q, $resource, BarGraphService, MapGraphService, LinearGraphService, Oboe) {
+      Object.assign(this, {$q, $resource, BarGraphService, MapGraphService, LinearGraphService, Oboe});
+      this.tweeters = [];
+      this.streams = [];
+      this.words = [
+      { id: 1,
+        word: ''
+      }];
+    }
+
+    add(word, inputNumber) {
+
+      this.streams = [];
+      if (this.words.length < 4) {
+        this.words.push({
+          id: this.words.length + 1,
+          word: ''
+        });
+      }
+
+      var currentWords = [];
+      this.words.forEach((element) => {
+        currentWords.push(element.word);
+      });
+
+      var params = {
+        action: 'search',
+        query: word
+      }
+
+      this.tweets = this.$resource('/tweets/:action/:query', params);

```

Comment:
cala logike polaczenia powinnas wyseparowac do oddzielnego serwisu. kontroler nie powinien sie tym zajmowac.
kontroler ma:
. przechowywac Ci aktualny stan strony
. spinac rozne serwisy do operacji na danych

jest to zasada zgodna z paroma zasadami, ktorych powinnismy przestrzegac:
. stateless components
. SOLID
. podstawa MVC - thin controllers, fat services ( https://stackoverflow.com/questions/14044681/fat-models-and-skinny-controllers-sounds-like-creating-god-models )